### PR TITLE
Add email message editing page + previewing!

### PR DESCRIPTION
**Pull request by Aberdeener**
[Image](https://imgur.com/a/CDSDXlx)

Additionally:
- Moved all email related language terms into `email.php`
- Created function `Email::formatEmail(string $email, Language $language)`
- Created function `$language->set(string $file, string $term, string $value)`

Things to note:
- Only languages which have the `email.php` file and it is writable will appear in the dropdown to edit. This will not be an issue for people who download after this is merged.
- I may have missed some translations - sorry.

Extra features todo later:
- Support place holders in Email::formatEmail()

### CODE REVIEW

**Comment by Aberdeener** [comment]
Okay, added my last commit for this PR.

**Comment by WhenUHackUNoob** [comment]
hey sam when are you gonna sort this out it will be cool k ty :D

File: custom/panel_templates/Default/core/emails_edit_messages_preview.tpl [hunk 1]
```diff
@@ -0,0 +1,88 @@
+<html lang="en">
+
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Email Preivew</title>
```
Comment (samerton) [inline custom/panel_templates/Default/core/emails_edit_messages_preview.tpl:6]: Typo in "Preview"

File: modules/Core/pages/panel/emails.php [hunk 2]
```diff
@@ -38,167 +38,244 @@
...
-if(isset($_GET['action']) && $_GET['action'] == 'test'){
-	$smarty->assign(array(
-		'SEND_TEST_EMAIL' => $language->get('admin', 'send_test_email'),
-		'BACK' => $language->get('general', 'back'),
-		'BACK_LINK' => URL::build('/panel/core/emails')
-	));
+// Since emails are sent in the user's language, they need to be able to pick which language's messages to edit
+$editing_language = new Language(null, Session::get('editing_language') ?? $queries->getWhere('languages', array('is_default', '=', 1))[0]->name);
+$emails = array(
+	['register', 'Registration', ['subject' => $editing_language->get('emails', 'register_subject'), 'message' => $editing_language->get('emails', 'register_message')]],
```
Comment (samerton) [inline modules/Core/pages/panel/emails.php:44]: Are the English terms (eg `Registration`, `Change Password`) on these lines shown to the user? If so, should they be translated to display in the user's language?

File: modules/Forum/pages/forum/view_topic.php [hunk 3]
```diff
@@ -281,25 +281,104 @@
...
-								$user_following->user_id,
-								'new_reply',
-								array('path' => ROOT_PATH . '/modules/Forum/language', 'file' => 'forum', 'term' => 'new_reply_in_topic', 'replace' => array('{x}', '{y}'), 'replace_with' => array(Output::getClean($user->data()->nickname), Output::getClean($topic->topic_title))),
-								array('path' => ROOT_PATH . '/modules/Forum/language', 'file' => 'forum', 'term' => 'new_reply_in_topic', 'replace' => array('{x}', '{y}'), 'replace_with' => array(Output::getClean($user->data()->nickname), Output::getClean($topic->topic_title))),
-								URL::build('/forum/topic/' . $tid . '-' . $forum->titleToURL($topic->topic_title), 'pid=' . $last_post_id)
-							);
-							$queries->update('topics_following', $user_following->id, array(
-								'existing_alerts' => 1
-							));
+						//if($user_following->user_id != $user->data()->id){
```
Comment (samerton) [inline modules/Forum/pages/forum/view_topic.php:289]: Are the comments on these lines intentional?

File: modules/Forum/pages/forum/view_topic.php [hunk 4]
```diff
@@ -281,25 +281,104 @@
...
+								);
+								$queries->update('topics_following', $user_following->id, array(
+									'existing_alerts' => 1
+								));
+							//}
+							$user_info = $queries->getWhere('users', array('id', '=', $user_following->user_id));
+							array_push($users_following_info, ['email' => $user_info[0]->email, 'username' => $user_info[0]->username]);
+						//}
+					}
+					$path = join(DIRECTORY_SEPARATOR, array(ROOT_PATH, 'custom', 'templates', TEMPLATE, 'email', 'forum_topic_reply.html'));
```
Comment (samerton) [inline modules/Forum/pages/forum/view_topic.php:306]: We should probably allow users to choose whether they want to receive email notifications or not, maybe one for a future change.

Comment (Derkades) [reply modules/Forum/pages/forum/view_topic.php:306]: I think Aber was working on this for a different PR, see also #1618

File: modules/Core/pages/panel/emails.php (hunk 2 above)
Comment (Aberdeener) [reply modules/Core/pages/panel/emails.php:44]: Ah, that was silly. Thanks

File: modules/Forum/pages/forum/view_topic.php (hunk 3 above)
Comment (Aberdeener) [reply modules/Forum/pages/forum/view_topic.php:289]: Wow, how did I not notice that. I commented them out while testing.

File: modules/Forum/pages/forum/view_topic.php (hunk 4 above)
Comment (Aberdeener) [reply modules/Forum/pages/forum/view_topic.php:306]: Yeah, I was planning on that for another PR ^

**Pull request merged**